### PR TITLE
fix(curriculum): prevent submitting array instead of string

### LINKED ISSUE #40715: Remove spaces from beginning and end passed with result as array instead of string
**Describe your problem and how to reproduce it:**
the challenge accepts 
```
let hello = "   Hello, World!  ";
let wsRegex = /(?!=\s+)(?:\w+\D\s\w+\D)(?!=\s+)/; // Change this line
let result = hello.match(wsRegex); // Change this line
```

in this case though `result` has value of:
```
[ 'Hello, World!',
  index: 3,
  input: '   Hello, World!  ',
  groups: undefined ]
```

looking at the test, it is asserted as `assert(result == 'Hello, World!');`, with non strict equality
I suggest a change here

**Add a Link to the page with the problem:**

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/remove-whitespace-from-start-and-end

https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/remove-whitespace-from-start-and-end.md

**Tell us about your browser and operating system:**
* Browser Name: 
* Browser Version: 
* Operating System: 

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40715

### CODE REVIEW

**Review by gikf — COMMENTED** [review]
First and last test seems now to be checking basically the same thing, although first one still can give false results. Have you tried changing first test to compare with strict equality instead?

**Comment by nhcarrigan** [comment]
Yes, I can confirm that changing the first test to use the strict equality operator also resolves the linked issue.

**Review by gikf — DISMISSED** [review]
Looks good.

Solves issue with `result` not being string and with setting `Hello, World!` directly.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/remove-whitespace-from-start-and-end.md [hunk 1]
```diff
@@ -30,10 +30,10 @@ Your solution should not use the `String.prototype.trim()` method.
 assert(!code.match(/\.?[\s\S]*?trim/));
 ```
 
-The `result` variable should not be set equal to a string.
+The `result` variable should not directly be set to a string
 
 ```js
-assert(!code.match(/result\s*=\s*".*?"/));
+assert(!code.match(/result\s*=\s*["'].*?["']/));
```
Comment (moT01) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/remove-whitespace-from-start-and-end.md:36]:
```suggestion
assert(!code.match(/result\s*=\s*["'`].*?["'`]/));
```

**Review by moT01 — COMMENTED** [review]
One minor change, other than that LGTM 🎉

**Review by gikf — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @Sembauke 🎉 Thanks for contributing to freeCodeCamp 🎉

**Pull request merged**